src/Mod/PartDesign/Gui/TaskRevolutionParameters.cpp

```diff
@@ -56,18 +56,18 @@ TaskRevolutionParameters::TaskRevolutionParameters(PartDesignGui::ViewProvider*
     QMetaObject::connectSlotsByName(this);
     this->groupLayout()->addWidget(proxy);
 
-    //bind property mirrors
-    PartDesign::ProfileBased* pcFeat = static_cast<PartDesign::ProfileBased*>(vp->getObject());
+    // bind property mirrors
+    auto pcFeat = dynamic_cast<PartDesign::ProfileBased *>(vp->getObject());
     if (pcFeat->isDerivedFrom(PartDesign::Revolution::getClassTypeId())) {
-        PartDesign::Revolution* rev = static_cast<PartDesign::Revolution*>(vp->getObject());
+        auto *rev = dynamic_cast<PartDesign::Revolution *>(vp->getObject());
         this->propAngle = &(rev->Angle);
         this->propMidPlane = &(rev->Midplane);
         this->propReferenceAxis = &(rev->ReferenceAxis);
         this->propReversed = &(rev->Reversed);
         ui->revolveAngle->bind(rev->Angle);
     } else {
         assert(pcFeat->isDerivedFrom(PartDesign::Groove::getClassTypeId()));
-        PartDesign::Groove* rev = static_cast<PartDesign::Groove*>(vp->getObject());
+        auto *rev = dynamic_cast<PartDesign::Groove *>(vp->getObject());

```

Comment:
Same as above, no need to replace the static_assert. But this whole block should be written as:
```
...
else if (pcFeat->isDerivedFrom(PartDesign::Groove::getClassTypeId())) {
    auto* rev = static_cast<PartDesign::Groove*>(vp->getObject());
....
}
```

The alternative way would be dynamic_cast without the previous type check:
```
else if (auto *rev = dynamic_cast<PartDesign::Groove *>(vp->getObject())) {
...
}
```

Reply:
Since we only have 2 cases and check for a nullptr, I thought keeping the existing if...else is fine. Why do you prefere an else if here?

Reply:
Who gives you give the guarantee that a `PartDesign::ProfileBased` must either be a `PartDesign::Revolution` or a `PartDesign::Groove`? If for some reason the wrong type is passed to the constructor of TaskRevolutionParameters then it will cause a crash. So, ideally the constructor should throw a Base::TypeError if the wrong type is passed:
```
if (/*is revolution*/) {
...
}
else if (/*is groove*/) {
...
}
else {
    throw Base::TypeError("Add some useful text here");
}
```
Doing it this way requires one further small change: the creation of the proxy widget must be moved to the initialization list to avoid a memory leak.

Reply:
Thanks. I made this now, but I am not sure I did it right,